draft-ietf-quic-recovery.md

```diff
@@ -226,24 +226,25 @@ QUIC supports many ACK ranges, opposed to TCP's 3 SACK ranges.  In high loss
 environments, this speeds recovery, reduces spurious retransmits, and ensures
 forward progress without relying on timeouts.
 
-### Explicit Correction For Delayed ACKs
+### Explicit Correction For Delayed ACKs {#ack-delay}
 
-QUIC ACKs explicitly encode the delay incurred at the receiver between when a
-packet is received and when the corresponding ACK is sent.  This allows the
-receiver of the ACK to adjust for receiver delays, specifically the delayed ack
-timer, when estimating the path RTT.  This mechanism also allows a receiver to
-measure and report the delay from when a packet was received by the OS kernel,
-which is useful in receivers which may incur delays such as context-switch
-latency before a userspace QUIC receiver processes a received packet.
+An endpoint measures the delay incurred between when a packet is received and
+when the corresponding ACK is sent.  The endpoint encodes this host delay for
+the largest acknowledged packet in the Ack Delay field of an ACK frame (see
+Section 19.3 of {{QUIC-TRANSPORT}}).  This allows the receiver of the ACK to
+adjust for any host delays - importantly, for delayed acknowledgements - when
+estimating the path RTT.  Where possible, this delay can include the time that a
+packet was held in the OS kernel before being processed by a userspace QUIC
+stack.
 
+An endpoint MUST NOT excessively delay acknowledgements of ack-eliciting
```

Comment:
I think that the most important thing here is the contract that `max_ack_delay` implies: An endpoints promises to never delay acknowledgments of an ack-eliciting packet by more than the indicated value.  If they do, any excess accrues to the RTT estimate and could result in timeouts.

Reply:
Heh -- I used the word contract and then removed it. I'll use your words.